src/app/(homepage)/page.js

```diff
@@ -7,9 +7,18 @@ import Benefits from "@/app/(homepage)/Benefits";
 import SignUp from "@/app/(homepage)/SignUp";
 import Demo from "@/app/(homepage)/Demo";
 import Security from "@/app/(homepage)/Security";
-
+import { useEffect, useContext } from 'react';
+import { useRouter } from "next/navigation";
+import { AuthTokenContext } from "@/app/layout";
 
 export default function Homepage() {
+  const router = useRouter();
+  const { setAuthToken } = useContext(AuthTokenContext);
```

Comment:
Why do you check for setAuthToken? It is a setter function. Is not it better to check for authToekn itself??